geopandas/tests/test_sindex.py

```diff
@@ -172,10 +181,299 @@ def test_intersection_invalid_bounds_tuple(self, test_geom):
                 # rtree raises an RTreeError which we need to catch
                 self.df.sindex.intersection(test_geom)
 
+    # ------------------------------ `query` tests ------------------------------ #
+    @pytest.mark.parametrize(
+        "predicate, test_geom, expected",
+        (
+            (None, box(-1, -1, -0.5, -0.5), []),
+            (None, box(-0.5, -0.5, 0.5, 0.5), [0]),
+            (None, box(0, 0, 1, 1), [0, 1]),
+            # bbox intersects but not geometry
```

Comment:
This comment doesn't seem to explain the following cases.  

It might make these easier to group mentally by the input geometries instead of by predicate, so that it is easier to see the variations in output based on just the predicate:

```
(None, LineString([(0, 1), (1, 0)]), [0, 1])  # bounding box intersects
("intersects", LineString([(0, 1), (1, 0)]), []),  # geometry does not intersect
("within", LineString([(0, 1), (1, 0)]), []),  # intersects but not within
("contains", LineString([(0, 1), (1, 0)]), []),  # intersects but not contains
```

Might be good to add a `touches` variant to this one too, but overall I agree with other comments that predicate testing does not need to be exhaustive here if appropriately covered already for rtree implementation (since most predicates are at least partly covered by pygeos tests).

Reply:
My comments were badly organized, that was supposed to refer to just the geometry under it. I added more comments and reorganized them.

I think I am going to skip reorganizing by input geometry. I do see the argument for it, but it would require reorganizing all test cases to be consistent. Let me know if this is okay.

Reply:
It is fine not to reorganize, so long as the comments make sense for the conditions.

Reply:
Ok. Let me know if the new comments help.